core/entity.ts

```diff
@@ -3,16 +3,18 @@ import {transform} from "./components/com_transform.js";
 import {Entity, Game} from "./game.js";
 import {Has, World} from "./world.js";
 
-const MAX_ENTITIES = 10000;
-
 export function create_entity(world: World) {
-    for (let i = 0; i < MAX_ENTITIES; i++) {
-        if (i === world.Signature.length || world.Signature[i] === 0) {
-            world.Signature[i] = 0;
-            return i;
-        }
+    if (world.Graveyard.length > 0) {
+        return world.Graveyard.pop()!;
+    }
+
+    if (world.Signature.length > 10000) {
+        throw new Error("No more entities available."); // DEBUG
     }
-    throw new Error("No more entities available.");
+
+    let entity = world.Signature.length;
+    world.Signature.push(0);
+    return entity;
```

Comment:
This could be simplified:
```suggestion
    return world.Signature.push(0) - 1;
```

Reply:
I'm not sure I'd call it _simpler_, but it's definitely shorter ;)